driver-sync/src/test/functional/com/mongodb/client/OidcAuthenticationProseTests.java

```diff
@@ -0,0 +1,937 @@
+ * limitations under the License.
+ */
+
+package com.mongodb.client;
+
+import com.mongodb.ConnectionString;
+import com.mongodb.MongoClientSettings;
+import com.mongodb.MongoCommandException;
+import com.mongodb.MongoConfigurationException;
+import com.mongodb.MongoCredential;
+import com.mongodb.MongoCredential.IdpResponse;
+import com.mongodb.MongoCredential.OidcRefreshCallback;
+import com.mongodb.MongoSecurityException;
+import com.mongodb.event.CommandListener;
+import com.mongodb.internal.connection.InternalStreamConnection;
+import com.mongodb.internal.connection.OidcAuthenticator;
+import com.mongodb.internal.connection.TestCommandListener;
+import com.mongodb.lang.Nullable;
+import org.bson.BsonArray;
+import org.bson.BsonBoolean;
+import org.bson.BsonDocument;
+import org.bson.BsonInt32;
+import org.bson.BsonString;
+import org.jetbrains.annotations.NotNull;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.function.Executable;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.CsvSource;
+import org.opentest4j.AssertionFailedError;
+import org.opentest4j.MultipleFailuresError;
+
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.NoSuchFileException;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.time.Duration;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Random;
+import java.util.concurrent.ConcurrentLinkedQueue;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.function.Supplier;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import static com.mongodb.MongoCredential.ALLOWED_HOSTS_KEY;
+import static com.mongodb.MongoCredential.IdpInfo;
+import static com.mongodb.MongoCredential.OidcRefreshContext;
+import static com.mongodb.MongoCredential.OidcRequestCallback;
+import static com.mongodb.MongoCredential.OidcRequestContext;
+import static com.mongodb.MongoCredential.PROVIDER_NAME_KEY;
+import static com.mongodb.MongoCredential.REFRESH_TOKEN_CALLBACK_KEY;
+import static com.mongodb.MongoCredential.REQUEST_TOKEN_CALLBACK_KEY;
+import static com.mongodb.MongoCredential.createOidcCredential;
+import static com.mongodb.client.TestHelper.setEnvironmentVariable;
+import static java.lang.System.getenv;
+import static java.util.Arrays.asList;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.junit.jupiter.api.Assertions.fail;
+import static org.junit.jupiter.api.Assumptions.assumeTrue;
+import static util.ThreadTestHelpers.executeAll;
+
+public class OidcAuthenticationProseTests {
```

Comment:
Let's document the location of the specification prose tests implemented in this class.